packages/auth-construct/src/construct.ts

```diff
@@ -350,6 +350,9 @@ export class AmplifyAuth
         phoneNumber: DEFAULTS.IS_REQUIRED_ATTRIBUTE.phoneNumber(phoneEnabled),
         ...(props.userAttributes ? props.userAttributes : {}),
       },
+      customAttributes: {
+        passwordless_sign_up: new cognito.StringAttribute({ mutable: true }),
+      },
```

Comment:
The problem with user attributes is if the customer disables `signUp` it will be change that will require to create a new user pool.

Reply:
This is going to be an interesting problem to solve then. Since this is "standalone auth construct" code, i.e. direct cdk users who only use "auth construct" and not opt into passwordless ("passwordless-auth-construct") shouldn't see that.

This might be a bit tricky since auth construct doesn't seem to have a away to define custom attributes as far as I can tell. Therefore I suggest to add TODO and create release blocker github issue to figure this out later before we merge this feature to main.

Reply:
As for immutability , we have similar problem with "domain" in this PR (reverted for now due to test failures, but it'll come back) https://github.com/aws-amplify/amplify-backend/pull/756/files#diff-0ae889ed8a9821b6bcb870a09d9f9570ae785235d7b7269524307f4f7bf4d319 .
(see comments around `domainPrefix?: string`).

Reply:
Also have we considered storing this in some other place like DDB table?
(passwordless auth already defines some ddb tables for this kind of data)